### PR TITLE
Convert to v2 addon

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
The process:
- `npx ember-addon-migrator@latest`
- fix issues as they come up, try to retain existing compatibility matrix

I do not believe there are any breaking changes here as `ember-autofocus-modifier` already required ember-auto-import@v2.

The thing _I'm_ unsure about is if ya'lls existing release process will still work with these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,98 +1,28 @@
...
-Your ember component has just been rendered. Have you ever wanted to focus an
-element in the DOM right after that? Like focusing this lonely input on your
-page? Or like focusing a full-screen modal's close button?
-
-This package wraps a modifier to autofocus an element in the DOM. It works for
-inputs and for other elements.
-
-As an alternative to installing this package, you might want to consider using
-[this snippet](https://github.com/qonto/ember-autofocus-modifier/blob/v0.0.1/addon/modifiers/autofocus.js).
+[Short description of the addon.]
```
Comment (dbendaou) [inline README.md:4]: I think we've lost the content of the previous README here 😄

File: test-app/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# test-app
+
+This README outlines the details of collaborating on this Ember application.
+A short introduction of this app could easily go here.
```
Comment (dbendaou) [inline test-app/README.md:4]:
```suggestion
This application is only used as a test application for `ember-autofocus-modifier`
```

wdyt?

File: package.json [hunk 3]
```diff
@@ -1,100 +1,36 @@
...
   "name": "ember-autofocus-modifier",
-  "version": "5.0.0",
-  "description": "The default blueprint for ember-cli addons.",
-  "keywords": [
-    "ember-addon"
-  ],
-  "publishConfig": {
-    "registry": "https://registry.npmjs.org/"
-  },
+  "version": "0.0.0",
```
Comment (dbendaou) [inline package.json:3]: Question: so this doesn't affect the version of the package since it's located in `addon/package.json` right?

**Comment by dbendaou** [comment]
Thanks for the help here @NullVoxPopuli 
We will review it on Thursday and try to fix the release process if needed at the same time

**Comment by dbendaou** [comment]
@SkoebaSteve will publish a pre-release so we can try on our app before publishing the version

File: ember-autofocus-modifier/package.json [hunk 4]
```diff
@@ -60,11 +60,14 @@
   "ember-addon": {
     "version": 2,
     "type": "addon",
-    "main": "addon-main.cjs"
+    "main": "addon-main.cjs",
+    "app-js": {
+      "./modifiers/autofocus.js": "./dist/_app_/modifiers/autofocus.js"
+    }
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply ember-autofocus-modifier/package.json:66]: This is auto-generated by the `@embroider/addon-dev`'s rollup plugins and it's required for anyone relying on non-strict-mode / globally available things (allowing you to use `{{autofocus}}` without importing in a loose-mode template)

File: test-app/config/ember-cli-update.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "schemaVersion": "1.0.0",
+  "packages": [
+    {
+      "name": "ember-cli",
+      "version": "4.10.0-beta.0",
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply test-app/config/ember-cli-update.json:6]: aye, yeah this isn't needed :upside_down_face: 
was using an old version of the migrator

Comment (NullVoxPopuli) [inline test-app/config/ember-cli-update.json:6]:
```suggestion
      "version": "4.11.0",
```

File: package.json (hunk 3 above)
Comment (NullVoxPopuli) [reply package.json:3]: correct

File: ember-autofocus-modifier/rollup.config.mjs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    // to leave alone and keep in the published output.
+    addon.keepAssets(['**/*.css']),
+
+    // Remove leftover build artifacts when starting a new build.
+    addon.clean(),
+
+    // Copy Readme and License into published package
+    copy({
+      targets: [
+        { src: '../README.md', dest: '.' },
```
[the remark this answers is not in the document]
Comment (herzzanu) [reply ember-autofocus-modifier/rollup.config.mjs:53]: What would be the benefit of keeping it?

File: ember-autofocus-modifier/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (dbendaou) [reply ember-autofocus-modifier/package.json:66]: Ok nice, thanks for the explanation!

File: ember-autofocus-modifier/rollup.config.mjs (hunk 6 above)
[the remark this answers is not in the document]
Comment (dbendaou) [reply ember-autofocus-modifier/rollup.config.mjs:53]: I guess to directly look up the CHANGELOG in the node_modules, some package does it (`ember-intl`, `ember-api-actions`), some doesn't

[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply ember-autofocus-modifier/rollup.config.mjs:53]: It's default in the blueprint -- and some private renovate / dependabot instances don't have access to GitHub to pull the changelog, so including it in the package helps out those scenarios

File: ember-autofocus-modifier/package.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    "lint": "concurrently 'npm:lint:*(!fix)' --names 'lint:'",
+    "lint:fix": "concurrently 'npm:lint:*:fix' --names 'fix:'",
+    "lint:hbs": "ember-template-lint . --no-error-on-unmatched-pattern",
+    "lint:js": "eslint . --cache",
+    "lint:hbs:fix": "ember-template-lint . --fix --no-error-on-unmatched-pattern",
+    "lint:js:fix": "eslint . --fix",
+    "start": "rollup --config --watch",
+    "test": "echo 'A v2 addon does not have tests, run tests in test-app'",
+    "prepack": "yarn run build",
+    "prepare": "yarn run build"
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply ember-autofocus-modifier/package.json:29]: building the addon is required in order to run tests, so this helps out with folks cloning the repo for the first time when they have work to do in the test-app.

turbo/nx can automate these dependent scripts, but I'm still working out some things (mostly with pnpm, which this repo isn't using, so :shrug: lol )

**Pull request merged**